src/cdk/api/subRoute.ts

```diff
@@ -1,31 +1,45 @@
+import { HttpApi } from "@aws-cdk/aws-apigatewayv2"
+import { LambdaProxyIntegration } from "@aws-cdk/aws-apigatewayv2-integrations"
 import { Construct } from "@aws-cdk/core"
-import { ApiView, ApiViewMixin, IEndpoint } from "./api"
+import { IApiViewClassMetadata } from "../../metadata"
+import { ApiViewMixin, IEndpoint } from "./api"
 
-export interface ISubRouteApiProps extends Omit<IEndpoint, "httpApi"> {
-  parentApi: ApiView
-  path: string
+export interface SubRouteApiProps extends Omit<IEndpoint, "httpApi"> {
+  lambdaApiIntegration: LambdaProxyIntegration
+  parentApiMeta: IApiViewClassMetadata
+  parentPath: string
+  path?: string
+  httpApi: HttpApi
 }
 
 /**
- * Route chained off an existing route and function.
+ * Route chained off an existing path and function.
  *
  * @category Construct
  */
 export class SubRouteApi extends ApiViewMixin {
-  constructor(scope: Construct, id: string, { methods, parentApi, path }: ISubRouteApiProps) {
+  constructor(
+    scope: Construct,
+    id: string,
+    { path, parentPath, unauthorized, authorizationScopes, parentApiMeta, ...rest }: SubRouteApiProps
+  ) {
     super(scope, id)
 
     // join parent path and our path together
-    if (!path.startsWith("/")) path = `/${path}`
-    path = parentApi.path + path
+    if (path && !path.startsWith("/")) path = `/${path}`
+    path = parentPath + (path || "")
+
+    // inherit parent `unauthorized` and `authorizationScopes`
+    if (typeof unauthorized === "undefined") unauthorized ||= parentApiMeta?.unauthorized
+    if (typeof authorizationScopes === "undefined") authorizationScopes ||= parentApiMeta?.authorizationScopes
```

Comment:
why not just `unauthorized === undefined`?
also this is kinda hard to read.

Reply:
yeah i think you're right it can be just `unauthorized === undefined`. i just don't want to override the value if it was explicitly set.